adapters/repos/db/shard_write_inverted_lsm.go

```diff
@@ -51,11 +53,66 @@ func (s *Shard) extendInvertedIndicesLSM(props []inverted.Property,
 				}
 			}
 		}
+
+		// add non-nil properties to the null-state inverted index, but skip internal properties (__meta_count, _id etc)
+		if (len(prop.Name) > 12 && prop.Name[len(prop.Name)-12:] == "__meta_count") ||
+			prop.Name[0] == '_' ||
+			!s.index.invertedIndexConfig.IndexNullState {
+			continue
+		}
+
+		if err := s.addIndexedNullStateToProps(docID, prop.Name, false); err != nil {
+			return errors.Wrap(err, "add indexed null state")
+		}
+	}
+
+	// add nil properties to the nullstate inverted index
+	if s.index.invertedIndexConfig.IndexNullState {
+		for _, nilProp := range nilProps {
+			if err := s.addIndexedNullStateToProps(docID, nilProp, true); err != nil {
+				return errors.Wrap(err, "add indexed null state")
+			}
+		}
 	}
 
 	return nil
 }
 
+func (s *Shard) addIndexedNullStateToProps(docID uint64, propName string, isNil bool) error {
+	bNullState := s.store.Bucket(helpers.BucketFromPropNameLSM(propName + filters.InternalNullIndex))
+	if bNullState == nil {
+		return errors.Errorf("no bucket for nil prop '%s' found", propName+filters.InternalNullIndex)
+	}
+
+	hashBucketNullState := s.store.Bucket(helpers.HashBucketFromPropNameLSM(propName + filters.InternalNullIndex))
+	if bNullState == nil {
+		return errors.Errorf("no nil-hash bucket for prop '%s' found", propName+filters.InternalNullIndex)
+	}
+
+	if bNullState.Strategy() != lsmkv.StrategySetCollection {
+		panic("prop has no frequency, but bucket does not have 'Set' strategy")
```

Comment:
I am wondering if it is worth panicking at this place!

Reply:
that's copy/pasted from the other functions that write to the II (adapters/repos/db/shard_write_inverted_lsm.go::153 for exmaple). I can change it to returning an error, but there was probably a reason for panicking in these cases

Reply:
I cannot tell if a panic is ok at this place. 
However if you are sure about it then it is fine with me too.